driver/iterator.go

```diff
@@ -117,3 +119,64 @@ func (i *eventReadIterator) Close() {
 	_ = i.close()
 	i.eof = true
 }
+
+type SearchResultIterator interface {
+	Next(r *SearchResponse) bool
+	Err() error
+	Close()
+}
+
+type searchStreamReader interface {
+	read() (*SearchResponse, error)
+	close() error
+}
+
+type searchResultIterator struct {
+	searchStreamReader
+	eof bool
+	err error
+}
+
+func (i *searchResultIterator) Next(r *SearchResponse) bool {
+	if i.eof {
+		return false
+	}
+
+	resp, err := i.read()
+	if err == io.EOF {
+		i.eof = true
+		_ = i.close()
+		return false
+	}
+	if err != nil {
+		i.eof = true
+		i.err = err
+		_ = i.close()
+		return false
+	}
+
+	if resp == nil {
+		*r = SearchResponse{}
+	} else {
+		// avoiding copy locks
```

Comment:
Any better way to avoid `sync.mutex locks being copied` warning when assigning `r = resp`?

Reply:
Yes, I think we can avoid copying by defining SearchResponse as a pointer below.
Like this `type SearchResponse *api.SearchResponse`